docs/design/code_and_name_organization/README.md

```diff
@@ -671,6 +673,51 @@ namespace Shapes;
 struct Shapes.Square { ... };
 ```
 
+#### Declaring namespace members
+
+Namespace members may only be declared in the same name scope which was used to
+declare the namespace. For example:
+
+```carbon
+namespace NS;
+
+// ✅ Allowed: declaration is in file scope, which also declared `NS`.
+class NS.ClassT {
+  // ❌ Error: A class body has its own name scope.
+  var NS.a: i32 = 0;
+}
+
+fn Function() {
+  // ❌ Error: A function body has its own name scope.
+  var NS.b: i32 = 1;
+}
+
+// ✅ Allowed: declaration is in file scope, which also declared `NS`.
+namespace NS.MemberNS;
+
+// ✅ Allowed: declaration is in file scope, which also declared `NS.MemberNS`.
+class NS.MemberNS.MemberClassT {}
+```
+
+When there are multiple declared names in binding patterns, particularly `var`
+and `let`, all names must be in the same namespace. For example:
```

Comment:
I think we should specify the scope over which the rule applies. For example:

```suggestion
When there are multiple declared names in binding patterns in the same pattern, particularly `var`
and `let`, all names must be in the same namespace. For example:
```
or
```suggestion
When multiple names are declared by binding patterns in the same pattern, particularly `var`
and `let`, all names must be in the same namespace. For example:
```
or perhaps more simply (given that you mention binding patterns below)
```suggestion
When multiple names are declared in the same pattern, particularly `var`
and `let`, all names must be in the same namespace. For example:
```

(As written, this could be read as disallowing `var NS.a: i32; var NS2.b: i32;` -- there are multiple declared names in binding patterns there.)

I also think it'd be useful to say the "particularly" part more directly, and less as an aside. Maybe we could say something along these lines:

> Because namespace members can only be declared in the same scope as the namespace, a namespace-qualified pattern binding can only be used in the pattern of a `var` or `let` declaration.

(I think that using "particularly" like this in a language rule could be unclear: does it mean that the rule only applies in that case, or applies in that case more than in other cases? Here, I think what's happening is that the rule applies uniformly to all patterns, but there happen to only be two cases where it *can* apply currently, due to other rules.)

Reply:
Done, I think